WebHostLib/autolauncher.py

```diff
@@ -154,8 +154,10 @@ def keep_running():
                     while 1:
                         time.sleep(0.1)
                         with db_session:
+                            # for update locks the database row(s) during transaction, preventing writes from elsewhere
                             to_start = select(
-                                generation for generation in Generation if generation.state == STATE_QUEUED)
+                                generation for generation in Generation
+                                if generation.state == STATE_QUEUED).for_update()
```

Comment:
I honestly don't understand that traceback xD
I'm assuming the SELECT is lazy and that's why it throws in that specific line? But how would it be possible that a row was already modified at that point unless there is another issue?

Either way we can give it a shot, because I think it would fix the race I mentioned above. It may not be the ideal solution for that, but I don't think locking while queuing the workers is gonna be a bottleneck.